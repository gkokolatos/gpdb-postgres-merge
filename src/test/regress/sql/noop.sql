DROP TABLE IF EXISTS noop_test CASCADE;

CREATE TABLE noop_test (a integer, b integer, c text) DISTRIBUTED BY (a);
CREATE UNIQUE INDEX noop_test_b_key ON noop_test USING NOOP (b);
CREATE UNIQUE INDEX noop_test_c_key ON noop_test USING NOOP (c);
\d noop_test
\d noop_test_b_key
\d noop_test_c_key

SELECT
	c2.relname,
	i.indisprimary,
	i.indisunique,
	i.indisclustered,
	i.indisvalid,
	pg_catalog.pg_get_indexdef(i.indexrelid, 0, true),
	pg_catalog.pg_get_constraintdef(con.oid, true),
	contype,
	condeferrable,
	condeferred,
	i.indisreplident,
	c2.reltablespace
FROM
	pg_catalog.pg_class c,
	pg_catalog.pg_class c2,
	pg_catalog.pg_index i   LEFT JOIN
	pg_catalog.pg_constraint con
ON
(
	conrelid = i.indrelid AND
	conindid = i.indexrelid AND
	contype IN ('p','u','x')
)
WHERE
	(c2.relname = 'noop_test_b_key' OR
	c2.relname = 'noop_test_c_key') AND
	c.oid = i.indrelid AND
	i.indexrelid = c2.oid
ORDER BY
	i.indisprimary DESC,
	i.indisunique DESC,
	c2.relname
;

INSERT INTO noop_test (a, b, c) SELECT generate_series(0, 9) as a, 1 as b, 'one' as c;
SELECT gp_segment_id, count(b) as "b multiples", count(c) as "c multiples", b, c FROM noop_test GROUP BY 1, b, c;
