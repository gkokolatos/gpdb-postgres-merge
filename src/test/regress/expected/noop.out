DROP TABLE IF EXISTS noop_test CASCADE;
CREATE TABLE noop_test (a integer, b integer, c text) DISTRIBUTED BY (a);
CREATE UNIQUE INDEX noop_test_b_key ON noop_test USING NOOP (b);
CREATE UNIQUE INDEX noop_test_c_key ON noop_test USING NOOP (c);
\d noop_test
   Table "public.noop_test"
 Column |  Type   | Modifiers 
--------+---------+-----------
 a      | integer | 
 b      | integer | 
 c      | text    | 
Indexes:
    "noop_test_b_key" UNIQUE, noop (b)
    "noop_test_c_key" UNIQUE, noop (c)
Distributed by: (a)

\d noop_test_b_key
Index "public.noop_test_b_key"
 Column |  Type   | Definition 
--------+---------+------------
 b      | integer | b
unique, noop, for table "public.noop_test"

\d noop_test_c_key
Index "public.noop_test_c_key"
 Column | Type | Definition 
--------+------+------------
 c      | text | c
unique, noop, for table "public.noop_test"

SELECT
	c2.relname,
	i.indisprimary,
	i.indisunique,
	i.indisclustered,
	i.indisvalid,
	pg_catalog.pg_get_indexdef(i.indexrelid, 0, true),
	pg_catalog.pg_get_constraintdef(con.oid, true),
	contype,
	condeferrable,
	condeferred,
	i.indisreplident,
	c2.reltablespace
FROM
	pg_catalog.pg_class c,
	pg_catalog.pg_class c2,
	pg_catalog.pg_index i   LEFT JOIN
	pg_catalog.pg_constraint con
ON
(
	conrelid = i.indrelid AND
	conindid = i.indexrelid AND
	contype IN ('p','u','x')
)
WHERE
	(c2.relname = 'noop_test_b_key' OR
	c2.relname = 'noop_test_c_key') AND
	c.oid = i.indrelid AND
	i.indexrelid = c2.oid
ORDER BY
	i.indisprimary DESC,
	i.indisunique DESC,
	c2.relname
;
     relname     | indisprimary | indisunique | indisclustered | indisvalid |                         pg_get_indexdef                         | pg_get_constraintdef | contype | condeferrable | condeferred | indisreplident | reltablespace 
-----------------+--------------+-------------+----------------+------------+-----------------------------------------------------------------+----------------------+---------+---------------+-------------+----------------+---------------
 noop_test_b_key | f            | t           | f              | t          | CREATE UNIQUE INDEX noop_test_b_key ON noop_test USING noop (b) |                      |         |               |             | f              |             0
 noop_test_c_key | f            | t           | f              | t          | CREATE UNIQUE INDEX noop_test_c_key ON noop_test USING noop (c) |                      |         |               |             | f              |             0
(2 rows)

INSERT INTO noop_test (a, b, c) SELECT generate_series(0, 9) as a, 1 as b, 'one' as c;
SELECT gp_segment_id, count(b) as "b multiples", count(c) as "c multiples", b, c FROM noop_test GROUP BY 1, b, c;
 gp_segment_id | b multiples | c multiples | b |  c  
---------------+-------------+-------------+---+-----
             1 |           2 |           2 | 1 | one
             0 |           5 |           5 | 1 | one
             2 |           3 |           3 | 1 | one
(3 rows)

